File: src/main/java/com/example/dpop/orchestrator/api/v1/identification/fsc/FscPending.java
package com.example.dpop.orchestrator.api.v1.identification.fsc;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public record FscPending(String kvnr, String fsc) {

    public static FscPending empty() {
        return new FscPending(null, null);
    }

    public FscPending merge(Map<String, Object> patch) {
        if (patch == null) return this;
        String newKvnr = patch.containsKey("kvnr") ? (String) patch.get("kvnr") : kvnr;
        String newFsc = patch.containsKey("fsc") ? (String) patch.get("fsc") : fsc;
        return new FscPending(newKvnr, newFsc);
    }

    public List<String> missingFields() {
        List<String> missing = new ArrayList<>();
        if (kvnr == null || kvnr.isBlank()) missing.add("kvnr");
        if (fsc == null || fsc.isBlank()) missing.add("fsc");
        return missing;
    }

    public boolean isComplete() { return missingFields().isEmpty(); }
}

File: src/main/kotlin/com/example/dpop/orchestrator/api/v1/identification/fsc/FscPending.kt
package com.example.dpop.orchestrator.api.v1.identification.fsc

@JvmRecord
data class FscPending(val kvnr: String?, val fsc: String?) {

    fun merge(patch: Map<String, Any>?): FscPending {
        if (patch == null) return this
        val newKvnr = if (patch.containsKey("kvnr")) patch["kvnr"] as String? else kvnr
        val newFsc = if (patch.containsKey("fsc")) patch["fsc"] as String? else fsc
        return FscPending(newKvnr, newFsc)
    }

    fun missingFields(): List<String> {
        val missing = mutableListOf<String>()
        if (kvnr.isNullOrBlank()) missing.add("kvnr")
        if (fsc.isNullOrBlank()) missing.add("fsc")
        return missing
    }

    val isComplete: Boolean get() = missingFields().isEmpty()

    companion object {
        @JvmStatic
        fun empty(): FscPending = FscPending(null, null)
    }
}
